core/src/proxy/block_engine_stage.rs

```diff
@@ -69,16 +71,20 @@ impl BlockEngineStageStats {
     }
 }
 
+
 pub struct BlockBuilderFeeInfo {
     pub block_builder: Pubkey,
     pub block_builder_commission: u64,
 }
 
 #[derive(Clone, Debug, Default, PartialEq, Eq)]
 pub struct BlockEngineConfig {
-    /// Block Engine URL
+    /// Primary Block Engine URL
     pub block_engine_url: String,
 
+    /// Secondary block engine URLs
+    pub secondary_block_engine_urls: Vec<String>,
```

Comment:
These are not ever mutated right? What if we instead of putting the secondary urls in the mutex we just pass in `block_engine_config: Either<Arc<Mutex<BlockEngineConfig>>, String>` (either: https://crates.io/crates/either). `Either::Right(String)` implies secondary so we can get rid of secondary boolean.